https://developer.berlios.de/patch/?func=detailpatch&patch_id=3501&group_id=5358
This needs to be fixed upstream.
The first three patches are safe, but I'm not sure how to properly fix
the last one to achieve the desired functionality.
The constants are already deprecated on 2.8.
--- src/plugins/contrib/wxContribItems/KWIC/include/wx/KWIC/AngularRegulator.h.orig
+++ src/plugins/contrib/wxContribItems/KWIC/include/wx/KWIC/AngularRegulator.h
@@ -11,6 +11,7 @@
 //
 /////////////////////////////////////////////////////////////////////////////
 
+#include <wx/control.h>
 
 const int kwxEVT_ANGREG_FIRST = wxEVT_FIRST + 5402;
 
--- src/plugins/contrib/wxContribItems/KWIC/include/wx/KWIC/LinearRegulator.h.orig
+++ src/plugins/contrib/wxContribItems/KWIC/include/wx/KWIC/LinearRegulator.h
@@ -11,6 +11,7 @@
 //
 /////////////////////////////////////////////////////////////////////////////
 
+#include <wx/control.h>
 
 const int kwxEVT_LINEARREG_FIRST = wxEVT_FIRST + 5401;
 
--- src/plugins/contrib/wxContribItems/wxTreeList/src/treelistctrl.cpp.orig
+++ src/plugins/contrib/wxContribItems/wxTreeList/src/treelistctrl.cpp
@@ -49,8 +49,12 @@
 #include <wx/hashmap.h>
 
 #ifdef __WXMAC__
+#if wxCHECK_VERSION(2,9,0)
+#include "wx/osx/private.h"
+#else
 #include "wx/mac/private.h"
 #endif
+#endif
 
 #include "wx/treelistctrl.h"
 
@@ -2017,8 +2021,6 @@
                                    const wxString& name) {
 
 #ifdef __WXMAC__
-    if (style & wxTR_HAS_BUTTONS) style |= wxTR_MAC_BUTTONS;
-    if (style & wxTR_HAS_BUTTONS) style &= ~wxTR_HAS_BUTTONS;
     style &= ~wxTR_LINES_AT_ROOT;
     style |= wxTR_NO_LINES;
 
